lib/srv/db/sqlserver/engine.go

```diff
@@ -131,12 +131,12 @@ func (e *Engine) HandleConnection(ctx context.Context, sessionCtx *common.Sessio
 func (e *Engine) receiveFromClient(clientConn, serverConn io.ReadWriteCloser, clientErrCh chan<- error, sessionCtx *common.Session) {
 	defer func() {
 		if r := recover(); r != nil {
-			e.Log.Warnf("Recovered while handling DB connection %v", r)
+			e.Log.ErrorContext(e.Context, "Recovered while handling DB connection", "recover", r)
```

Comment:
Warn turned into Error?

```suggestion
			e.Log.WarnContext(e.Context, "Recovered while handling DB connection", "recover", r)
```

Reply:
Yes, that one is intentional. I think panics deserve higher severity than warn.